dotCMS/src/main/java/com/dotmarketing/portlets/cmsmaintenance/factories/CMSMaintenanceFactory.java

```diff
@@ -17,138 +18,175 @@
+     * newest version. If no date is found, the current date will be returned.
+     *
+     * @return The oldest modification date from a versionable object.
+     */
+    private static Calendar getOldestInodeDate() {
+        Calendar startDate = Calendar.getInstance();
+        final DotConnect dc = new DotConnect();
+        dc.setSQL(
+                "SELECT MIN(idate) AS min_date FROM inode WHERE type IN ('contentlet', 'containers', 'template')");
+        dc.setMaxRows(1);
+        try {
+            final List<Map<String, Object>> results = dc.loadObjectResults();
+            final Date minDate = (Date) results.get(0).get("min_date");
+            if (null != minDate) {
+                startDate.setTime(minDate);
+            }
+        } catch (final DotDataException e) {
+            Logger.error(CMSMaintenanceFactory.class, String.format("An error occurred when finding the minimum date." +
+                    " Using date specified by user instead: '%s'", e.getMessage()), e);
+            startDate = null;
+        }
+        return startDate;
+    }
 
-		/*
-		 * Run the drop tasks interatively, moving forward in time 
-		 * DROP_OLD_ASSET_ITERATE_BY_SECONDS controls how many seconds to
-		 * move forward in time for each iteration - default is to iterate by 30 days
-		 */
-		Calendar runDate = Calendar.getInstance();
-		runDate.setTime(assetsOlderThan);
-		runDate.add(Calendar.YEAR, -2);
-		runDate.set(Calendar.HOUR_OF_DAY, 0);
-		runDate.set(Calendar.MINUTE, 0);
-		runDate.set(Calendar.SECOND, 0);
-		runDate.set(Calendar.MILLISECOND, 0);
+    /**
+     * Deletes all the versions of dotCMS objects that are older than the specified
+     * date.
+     *
+     * @param assetsOlderThan The date up to which object versions will be deleted.
+     *
+     * @return The total The total number of versions that were deleted for the specified date.
+     *
+     * @throws DotDataException An error occurred when deleting the data.
+     */
+    private static int removeOldVersions(final Calendar assetsOlderThan) throws DotDataException {
+        int deletedRecords = 0;
+        int totalRecords = 0;
+        Logger.info(CMSMaintenanceFactory.class, String.format("-> Dropping versions older than '%s':",
+                UtilMethods.dateToHTMLDate(assetsOlderThan.getTime(), "yyyy-MM-dd")));
 
-        try{
-    		DotConnect dc = new DotConnect();
-            String minIdateSQL = "select idate from inode order by idate";
-            
-            dc.setSQL(minIdateSQL);
-            dc.setMaxRows(1);
-        	List<Map<String, Object>> map =  dc.loadObjectResults();
-        	Date d = (Date) map.get(0).get("idate");
-        	if(d !=null)
-        		runDate.setTime(d);
+        deletedRecords = APILocator.getContentletAPI().deleteOldContent(assetsOlderThan.getTime());
+        if (deletedRecords > 0) {
+            Logger.info(CMSMaintenanceFactory.class, " Removed " + deletedRecords + " old Contentlets");

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal " Removed " appears 5 times in this file; the first occurrence is on line 154](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5534906)

Reply:
Fixed.